web/src/code-editor.ts

```diff
@@ -0,0 +1,110 @@
+import { LitElement, html, css } from "lit";
+import { customElement, queryAsync, state } from "lit/decorators.js";
+
+import { closeBrackets } from "@codemirror/autocomplete";
+import { history, emacsStyleKeymap } from "@codemirror/commands";
+import { bracketMatching, indentOnInput } from "@codemirror/language";
+import { EditorState } from "@codemirror/state";
+import { EditorView, keymap, highlightActiveLine } from "@codemirror/view";
+
+import { QueryResultPayload, rawQuery } from "./api";
+
+@customElement("code-editor")
+export class CodeEditor extends LitElement {
+  static styles = css`
+    :host {
+      display: block;
+    }
+    #editor {
+      height: 100%;
+    }
+    .cm-editor {
+      border: 1px solid var(--sl-panel-border-color);
+      max-height: 100%;
+    }
+    sl-button {
+      width: 100%;
+      margin-bottom: 1rem;
+    }
+  `;
+
+  @queryAsync("#editor") _editor: Promise<HTMLElement>;
+  @state() _waiting = false;
+  #editor?: EditorView;
+
+  connectedCallback() {
+    super.connectedCallback();
+    this.addEditor();
+  }
+  async addEditor() {
+    const editor = await this._editor;
+    let startState = EditorState.create({
+      doc: `g.V().filter{
+  it.get()
+    .value('pname')
+    .matches('pspg')
+}.repeat(
+   outE()
+  .otherV()
+  .simplePath())
+.times(2)
+.path()
+.by('pname')
+.by(label)`,
```

Comment:
add an indication of what it does or what needs to be changed here to customize it?

Reply:
right, I didn't realize that we can have comments in the query